ext/liquid_c/block.c

```diff
@@ -262,6 +282,112 @@ static tag_markup_t internal_block_body_parse(block_body_t *body, parse_context_
     return unknown_tag;
 }
 
+VALUE parse_single_binary_comparison(VALUE markup) {
+    if (NIL_P(markup))
+        return Qnil;
+
+    StringValue(markup);
+    char *start = RSTRING_PTR(markup);
+
+    parser_t p;
+    init_parser(&p, start, start + RSTRING_LEN(markup));
+    VALUE a = internal_expression_parse(&p);
+    lexer_token_t op = parser_consume(&p, TOKEN_COMPARISON);
+    
+    if(op.type) {
+        VALUE op_str = rb_enc_str_new(op.val, op.val_end - op.val, utf8_encoding);
+        VALUE b = internal_expression_parse(&p);
+        return rb_funcall(cLiquidCondition, intern_new, 3, a, op_str, b);
+    } 
+
+    return rb_funcall(cLiquidCondition, intern_new, 1, a);
+}
+
+tag_markup_t parse_if_tag(VALUE markup, block_body_t *body, parse_context_t *parse_context) {
```

Comment:
theres no validation yet for guarding against badly structured if, for example an elsif without if.